src/inference/src/dev/core_impl.cpp

```diff
@@ -1076,8 +1076,20 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
         // explicitly enabled Auto-Batching
         auto pos = deviceName.find_first_of(":");
         if (pos == std::string::npos)
-            return model;  // BATCH device is already configured via the config
-        deviceNameWithBatchSize = deviceName.substr(pos + 1);
+            return model;
+        // BATCH device is already configured via the config
+        // handle the hw plugin doesn't support optimal_batch_size & user doesn't assign the batch size, such as
+        // "BATCH:CPU" for this situation will assign the batch size to 1. ("BATCH:CPU" -> "BATCH:CPU(1)")
+        if (deviceName.find("(") == std::string::npos) {
+            auto supported_properties =
+                ICore::get_property(deviceName.substr(pos + 1, deviceName.find("(")), ov::supported_properties, {});
+            if (std::find(supported_properties.begin(), supported_properties.end(), ov::optimal_batch_size) ==
+                supported_properties.end())
+                deviceNameWithBatchSize = deviceName.substr(pos + 1) + "(1)";
+        } else {
```

Comment:
if batch = 1, we don't need to perform auto-batching here

Reply:
@ilya-lavrenov 
https://github.com/openvinotoolkit/openvino/blob/494a9cf9a9fc6d953df27d04dc0dcedb79cc7b93/src/plugins/auto_batch/src/plugin.cpp#L239-L240
GPU plugin can support OPTIMAL_BATCH_SIZE with model as the option of property OPTIMAL_BATCH_SIZE
Is it reasonable that CPU plugin supports the same, but return 1 for now?

Reply:
yes, it's reasonable. But we still can have plugin which don't support this property, hence we have to handle this situation on Core side.